src/System.Net.Sockets/src/System/Net/Sockets/BaseOverlappedAsyncResult.Windows.cs

```diff
@@ -95,14 +95,17 @@ private static unsafe void CompletionPortCallback(uint errorCode, uint numBytes,
                     }
                     else
                     {
+                        bool addRefedNativeOverlapped = false;
                         try
                         {
+                            asyncResult._nativeOverlapped.DangerousAddRef(ref addRefedNativeOverlapped);
```

Comment:
> And unless I'm missing something, you're not doing this when we pass the overlapped in general, e.g. to WSASend etc. So why just here?

You mean the ones I commented with  "// SafeHandle was just created in SetUnmanagedStructures"?  The comment was meant to highlight that the reason the "Dangerous" operation is actually safe is because the SafeHandle was just created and hasn't been given out to anyone, therefore no one could be disposing of it concurrently.